### PR TITLE
faqs: update based on early flux rollout on production systems

### PR DESCRIPTION
**Pull request by garlick**
The FAQ didn't have any categories so I added a General category for most everything we have now, and another for MPI, and added some new entries to the MPI section based on our fun with mvapich and openmpi this past week.

A couple of other entries were updated, and one on F58 encoding issues was added.

### CODE REVIEW

File: faqs.rst [hunk 1]
```diff
@@ -1,35 +1,60 @@
...
+*****************
+General Questions
+*****************
+
+What's with the fancy ƒ?
+========================
+
+Flux job IDs and their multiple encodings are described in
+`RFC 19 <https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_19.html>`_. The default F58 encoding assumes that your terminal session can
+handle UTF-8.
```
Comment (grondo) [inline faqs.rst:21]: I wonder if we should state something like: The UTF-8 `ƒ` character is used to denote a Flux jobid in F58 encoding when the current locale supports UTF-8 multi-byte characters. When the locale does not support UTF-8, then an ASCII `f` is used. The ASCII rendering of jobids can be forced by setting ``FLUX_F58_FORCE_ASCII=1`` in your environment.

Actually, the main point is that Flux will try to first determine if the current locale supports UTF-8 multi-byte characters before using `ƒ`. Maybe we just say that? Up to you.

**Review by grondo — APPROVED** [review]
This is great, thanks!

Just one suggestion inline you can take it or leave it.

File: faqs.rst (hunk 1 above)
Comment (garlick) [reply faqs.rst:21]: Great point.  I'll mention something to that effect.

Comment (garlick) [reply faqs.rst]: I forced an update - look OK?

Comment (grondo) [reply faqs.rst]: Yes, nice!

**Pull request merged**